server/apps/immich/src/api-v1/asset/asset.service.spec.ts

```diff
@@ -67,29 +113,65 @@ describe('AssetService', () => {
   });
 
   // Currently failing due to calculate checksum from a file
-  // it('create an asset', async () => {
-  //   const assetEntity = _getAsset();
+  it('create an asset', async () => {
+    const assetEntity = _getAsset_1();
 
-  //   assetRepositoryMock.create.mockImplementation(() => Promise.resolve<AssetEntity>(assetEntity));
+    assetRepositoryMock.create.mockImplementation(() => Promise.resolve<AssetEntity>(assetEntity));
 
-  //   const originalPath =
-  //     'upload/3ea54709-e168-42b7-90b0-a0dfe8a7ecbd/original/116766fd-2ef2-52dc-a3ef-149988997291/51c97f95-244f-462d-bdf0-e1dc19913516.jpg';
-  //   const mimeType = 'image/jpeg';
-  //   const createAssetDto = _getCreateAssetDto();
-  //   const result = await sui.createUserAsset(authUser, createAssetDto, originalPath, mimeType);
+    const originalPath = 'fake_path/asset_1.jpeg';
+    const mimeType = 'image/jpeg';
+    const createAssetDto = _getCreateAssetDto();
+    const result = await sui.createUserAsset(
+      authUser,
+      createAssetDto,
+      originalPath,
+      mimeType,
+      Buffer.from('0x5041E6328F7DF8AFF650BEDAED9251897D9A6241', 'hex'),
+    );
 
-  //   expect(result.userId).toEqual(authUser.id);
-  //   expect(result.resizePath).toEqual('');
-  //   expect(result.webpPath).toEqual('');
-  // });
+    expect(result.userId).toEqual(authUser.id);
+    expect(result.resizePath).toEqual('');
+    expect(result.webpPath).toEqual('');
+  });
 
   it('get assets by device id', async () => {
-    assetRepositoryMock.getAllByDeviceId.mockImplementation(() => Promise.resolve<string[]>(['4967046344801']));
+    const assets = _getAssets();
+
+    assetRepositoryMock.getAllByDeviceId.mockImplementation(() =>
+      Promise.resolve<string[]>(Array.from(assets.map((asset) => asset.deviceAssetId))),
+    );
 
-    const deviceId = '116766fd-2ef2-52dc-a3ef-149988997291';
+    const deviceId = 'device_id_1';
     const result = await sui.getUserAssetsByDeviceId(authUser, deviceId);
 
-    expect(result.length).toEqual(1);
-    expect(result[0]).toEqual('4967046344801');
+    expect(result.length).toEqual(2);
+    expect(result).toEqual(assets.map((asset) => asset.deviceAssetId));
```

Comment:
Same here, you could define a `const assetsResult = ["with", "hardCoded", "ids"]` and return in the the promise and use it here in the `expect`